### PR TITLE
Cleaned LPI Data

### PR DESCRIPTION
**Pull request by manishkr1754**
Updated README with LPI Data Cleaning Notebook Link
[Kaggle Notebook](https://www.kaggle.com/code/manishkr1754/lpi-data-cleaning-and-preprocessing/notebook)

1. Backlash to Underscore formatting
2. Cleaned Course_Level and Difficulty_Level
3. Added Duration in minutes
4. Cleaned Keywords-Replaced backslash with comma
5. Updated MetaData

### CODE REVIEW

**Comment by manishkr1754** [comment]
Updated the Cleaned Data in Learning Path Index Excel sheet as well.

**Review by neomatrix369 — COMMENTED** [review]
In principle this looks good but the moment we update both the data and metadata on the Kaggle Dataset - do we know if the metadata will be reflected on the Kaggle Dataset description?

**Review by neomatrix369 — CHANGES_REQUESTED** [review]
Also, add reference(s) about the collaboration between @manishkr1754 and @entzyeung wrt to cleaning and preprocessing/postprocessing of data and creating the final outputs

**Comment by manishkr1754** [comment]
> In principle this looks good but the moment we update both the data and metadata on the Kaggle Dataset - do we know if the metadata will be reflected on the Kaggle Dataset description?

No, We need to manually enter the metadata info from the json file

**Comment by manishkr1754** [comment]
> Also, add reference(s) about the collaboration between @manishkr1754 and @entzyeung wrt to cleaning and preprocessing/postprocessing of data and creating the final outputs

For referencing collaborative work, we have shared the kaggle notebook link.

**Comment by neomatrix369** [comment]
> > Also, add reference(s) about the collaboration between @manishkr1754 and @entzyeung wrt to cleaning and preprocessing/postprocessing of data and creating the final outputs
> 
> For referencing collaborative work, we have shared the kaggle notebook link.

Both your and his notebooks links are referred to in the README?

let me make the changes if you dont follow what I mean

**Comment by neomatrix369** [comment]
> > In principle this looks good but the moment we update both the data and metadata on the Kaggle Dataset - do we know if the metadata will be reflected on the Kaggle Dataset description?
> 
> No, We need to manually enter the metadata info from the json file

If this is not a lot of changes are you happy to copy them manually from the .json onto the dataset description - assuming we wont be doing this often.

File: data/Learning_Pathway_Index.csv [hunk 1]
```diff
@@ -1,1398 +1,1397 @@
...
-Data Engineer - Google Cloud Big Data and Machine Learning Fundamentals,Google Cloud Skill Boost,Beginner,Free during mentorship period,Data Engineering for Streaming Data: Summary,1 minute,Medium,"data engineer, streaming data, Google Cloud, Big data, Machine Learning",https://www.cloudskillsboost.google/course_sessions/5007143/video/383348
-Data Engineer - Google Cloud Big Data and Machine Learning Fundamentals,Google Cloud Skill Boost,Beginner,Free during mentorship period,Data Engineering for Streaming Data: Quiz,2 minutes,Medium,"data engineer, streaming data, quiz, Google Cloud, Big data, Machine Learning",https://www.cloudskillsboost.google/course_sessions/5007143/quizzes/383349
+Course_Learning_Material,Source,Course_Level,Type,Module,Difficulty_Level,Keywords,Links,Duration_in_minutes
```
Comment (neomatrix369) [inline data/Learning_Pathway_Index.csv:1]: We do not need a `Duration_in_minutes` field the Duration should be human readable, for EDA purposes you can create a runtime/temp field if you like to do some stats on it - pls re-introduce the original Duration field and corresponding data

Comment (neomatrix369) [reply data/Learning_Pathway_Index.csv:1]: `Duration` isnt a numeric field, its a string and also can be expresses in minutes, hours or days where applicable.

**Comment by neomatrix369** [comment]
While we are at renaming columns,  would also like to add a Tab_Name column per submodule in the `Learning Path Index.csv` file if it's not already there. 

For each sub-module a corresponding parent table's tab name should be added, maybe even rename the field `Tab Name` to `Module Id` or `Module Code` and use this to link the two datasets - this will also help the LLM embeddings.

@manishkr1754

**Comment by manishkr1754** [comment]
> > > Also, add reference(s) about the collaboration between @manishkr1754 and @entzyeung wrt to cleaning and preprocessing/postprocessing of data and creating the final outputs
> > 
> > 
> > For referencing collaborative work, we have shared the kaggle notebook link.
> 
> Both your and his notebooks links are referred to in the README?
> 
> let me make the changes if you dont follow what I mean

My Kaggle notebook is inclusive of his work as well. Also, in future if he wants to add something, he can add in the same single notebook.

**Comment by manishkr1754** [comment]
> > > In principle this looks good but the moment we update both the data and metadata on the Kaggle Dataset - do we know if the metadata will be reflected on the Kaggle Dataset description?
> > 
> > 
> > No, We need to manually enter the metadata info from the json file
> 
> If this is not a lot of changes are you happy to copy them manually from the .json onto the dataset description - assuming we wont be doing this often.

It's fine. There are only about 10 fields and we have metadata json ready justwe have to copy and paste

**Comment by manishkr1754** [comment]
@neomatrix369,

I wanted to inform you that I have made the following updates in response to the Change Request:

1. Renamed the "Tab_Name" to "Module_Code"
2. Added the "Module_Code" column in the Learning Path Index Dataset.
3. Standardized the entries in the "Duration" column to represent minutes and hours in datetime format.
4. For Generative AI courses, I noticed that there were no "Module_Code" values in the Excel file. I have created two new module codes in the Course and Learning Material Dataset. These codes will need to be updated in the Excel file, which I understand is currently inaccessible to me.
5. Updated the MetaData information file as per the above changes.
6. Made updates to the Data Cleaning Kaggle Notebook, which can be found [LPI Data Cleaning Notebook](https://www.kaggle.com/code/manishkr1754/lpi-data-cleaning-and-preprocessing).

Once this merge is successfully completed, I will proceed to update the Excel and Kaggle Dataset accordingly.

**Review by neomatrix369 — APPROVED** [review]

**Pull request merged**